GeocodingLambda/integration.test.ts

```diff
@@ -0,0 +1,260 @@
+      }
+    },
+    placemark: {
+      type: "placemark",
+      value: {
+        city: "Sydney Olympic Park",
+        isoCountryCode: "AUS",
+        name: "Bennelong Pkwy, Sydney Olympic Park NSW 2127, Australia",
+        postalCode: "2127",
+        region: "New South Wales",
+        street: "Bennelong Pkwy"
+      }
+    }
+  }
+]
+
+const unknownAddressTestLocation: TestLocation = {
+  name: "Point Nemo (Pacific Ocean)",
+  coordinate: {
+    type: "coordinate",
+    value: {
+      latitude: -48.876667,
+      longitude: -123.393333
+    }
+  },
+  placemark: {
+    type: "placemark",
+    value: {}
+  }
+}
+
+describe("Geocoding lambda tests", () => {
+  const mockGeocoding = jest.fn()
+  const mockReverseGeocoding = jest.fn()
+
+  beforeEach(async () => {
+    await resetDB()
+  })
+
+  describe("Forward geocoding (address to coordinates)", () => {
+    it.each(testLocations)(
+      "Should find the coordinates and save a placemark given the address of $name",
+      async ({ placemark, coordinate }) => {
+        const result = await handler(placemark)
+
+        expect(result).toMatchObject({
+          coordinate: {
+            latitude: expect.closeTo(coordinate.value.latitude),
+            longitude: expect.closeTo(coordinate.value.longitude)
+          },
+          placemark: expect.objectContaining(
+            placemark.value
+          )
+        })
+
+        const cachedResult = await handler(placemark, mockGeocoding, mockReverseGeocoding)
+        expect(mockGeocoding).toHaveBeenCalledTimes(0)
+        expect(mockReverseGeocoding).toHaveBeenCalledTimes(0)
+
+        expect(cachedResult).toMatchObject({
+          coordinate: {
+            latitude: expect.closeTo(coordinate.value.latitude),
+            longitude: expect.closeTo(coordinate.value.longitude)
+          },
+          placemark: expect.objectContaining(
+            placemark.value
+          )
+        })
+      }
+    )
+
+    it("Should use default coordinates given an unknown address", async () => {
+      const result = await handler(unknownAddressTestLocation.placemark)
+
+      expect(result).toMatchObject({
+        coordinate: {
+          latitude: -90,
+          longitude: 0
+        }
+      })
+
+      expect(result.placemark).toStrictEqual({})
+    })
+  })
+
+  describe("Reverse geocoding (coordinates to address)", () => {
+    it.each(testLocations)(
+      "Should find the address and save a placemark given the coordinates of $name",
+      async ({ coordinate, placemark }) => {
+        const result = await handler(coordinate)
+
+        expect(result).toMatchObject({
+          coordinate: {
+            latitude: expect.closeTo(coordinate.value.latitude),
+            longitude: expect.closeTo(coordinate.value.longitude)
+          },
+          placemark: expect.objectContaining(
+            placemark.value
+          )
+        })
+
+        const cachedResult = await handler(coordinate, mockGeocoding, mockReverseGeocoding)
```

Comment:
Nit: For clarity, you could explicitly mock the mock functions to reject with a random error.